aspnetcore/fundamentals/dependency-injection.md

```diff
@@ -188,6 +188,19 @@ Observe which of the `OperationId` values vary within a request, and between req
 
 * *Singleton* objects are the same for every object and every request (regardless of whether an instance is provided in `ConfigureServices`)
 
+## Scope validation
+
+When the app is running in the Development environment on ASP.NET Core 2.0 or later, the default service provider performs checks to verify that:
+
+* Scoped services aren't directly or indirectly resolved from the root service provider.
+* Scoped services aren't directly or indirectly injected into singletons.
+
+The root service provider is created when [BuildServiceProvider](/dotnet/api/microsoft.extensions.dependencyinjection.servicecollectioncontainerbuilderextensions.buildserviceprovider) is called. The root service provider's lifetime usually corresponds to the app/server's lifetime.
```

Comment:
Using the word "usually" here reveals uncertainty. In which situations would the lifetime not correspond to the app/server's lifetime?

Reply:
I presume from @pakrym's remarks that if the dev builds one or more additional "root" containers on-the-fly that it won't exist for the full lifetime of the app/server.

Reply:
We could go with something more explicit, like ...

> The root service provider's lifetime corresponds to the app/server's lifetime when the provider starts when the app starts and is disposed when the app shuts down.